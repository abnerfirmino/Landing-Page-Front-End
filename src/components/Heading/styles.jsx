import styled, { css } from 'styled-components';

// configurando o title size
const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};`,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};`,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};`,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

function mediaFont(theme) {
  return css`
    @media screen and ${theme.media.ltMedium} {
      font-size: ${theme.fonts.sizes.xlarge};
    }
  `;
}

// função do uppercase
const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colordark, size, uppercase }) => css`
    color: ${colordark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
