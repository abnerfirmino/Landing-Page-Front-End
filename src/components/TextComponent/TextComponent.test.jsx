import { describe, expect, it } from 'vitest';
import { screen } from '@testing-library/react';
import { TextComponent } from '.';

import { renderTheme } from '../../styles/render-theme';

describe('TextComponent', () => {
  it('should render the correct text', () => {
    renderTheme(
      <TextComponent>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit.
      </TextComponent>,
    );

    const text = screen.getByText(/Lorem ipsum, dolor/i);

    expect(text).toBeInTheDocument();
  });

  it('should have default styles', () => {
    const { container } = renderTheme(
      <TextComponent>Lorem ipsum.</TextComponent>,
    );
    const text = container.firstChild;

    expect(text).toMatchSnapshot();
  });
});
