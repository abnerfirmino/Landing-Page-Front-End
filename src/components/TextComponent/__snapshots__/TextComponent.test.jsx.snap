// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextComponent > should have default styles 1`] = `
.c0 {
  font-size: 2.4rem;
}

<div
  class="c0"
>
  <p>
    Lorem ipsum.
  </p>
</div>
`;
